app/main/services/process_request_json.py

```diff
@@ -16,9 +16,34 @@ def process_values_for_matching(values):
     return values
 
 
+def _ensure_value_list(json_string_or_list):
+    if isinstance(json_string_or_list, list):
+        return json_string_or_list
+    else:
+        return [json_string_or_list]
+
+
 def convert_request_json_into_index_json(request_json):
     index_json = {}
 
+    for transformation in TRANSFORM_FIELDS:
+        # Each transformation can generate new field values in "target field" when
+        # certain values are present in "field". The example use case is when
+        # we are adding parent categories, whenever any one of their subcategories
+        # is present.
+        source_field = transformation['field']
+        target_field = transformation.get('target_field') or source_field
+
+        if source_field in request_json:
+            source_values = _ensure_value_list(request_json[source_field])
+            source_values_set = set(source_values)
+            target_values = _ensure_value_list(request_json[target_field])
```

Comment:
It would be nice to not have to 'figure it out' any more than necessary which can be helped with accurate nomenclature. As I skim the code now it doesn't tell me what kind of transformation it does, either through variable names or comments, so it requires a lot more active thought to work it out. It'll be fine for us in Catalogues but a bit more opaque for anyone else.

Other than that, maybe have `source_field` rather than just `field`. More consistent with `target_field`, and you drop it into a variable called `source_field` anyway. This would be nice-to-have, but don't really mind and shouldn't stop you merging, if you particularly disagree.

👍 if we want to discuss this and refactor later.